src/SolidStateDetector/Semiconductor.jl

```diff
@@ -51,7 +51,10 @@ end
 function Semiconductor{T}(dict::AbstractDict, input_units::NamedTuple, outer_transformations) where T <: SSDFloat
 
     impurity_density_model = if haskey(dict, "impurity_density") 
-        ImpurityDensity(T, dict["impurity_density"], input_units)
+        hascorrections = haskey(dict["impurity_density"], "corrections")
+        impurity_density_scale = hascorrections && haskey(dict["impurity_density"]["corrections"], "scale") ? _parse_value(T, dict["impurity_density"]["corrections"]["scale"], NoUnits) : T(1)
```

Comment:
I know that we use this syntax `haskey(d, k) ? d[k] : default_value` a lot, which can be compactified using `get(d, k, default_value)`.

Maybe let's try to use it from now on to compactify code:
```suggestion
        impurity_density_scale = hascorrections ? _parse_value(T, get(dict["impurity_density"]["corrections"], "scale", 1), NoUnits) : T(1)
```
(and check if it behaves as intended)